### ISSUE TITLE
Changes are not reflected in the Sidebar when creating or deleting tags outside the Task Editor, until restarting the app

### ISSUE THREAD

**Issue by nekohayo** [getting-things-gnome/gtg#1005]
Since the GTK4 port:

* If you create a task like `test @brand_new_tag` via the quick-add bar at the bottom of the task browser, it will not get added to the sidebar until restarting the app. It works correctly if typing the tag within a Task Editor window, so you probably just need to reuse that code.
* Right-click Deleting a tag from the sidebar doesn't have any visible effect after the confirmation dialog (removal throughout the GUI only happens after app restart)

Related: #1016

**Comment by SqAtx** [comment]
Ha, I came here to report the same thing for Saved Searches. When you delete a saved search:

![image](https://github.com/getting-things-gnome/gtg/assets/4351932/1074da3f-a740-493f-98dc-07050bd9bc1d)

it's not reflected in the UI until you restart GTG. Both problems can probably be fixed at the same time. If not, let's create a new issue for the saved searches.

**Comment by nekohayo** [comment]
Indeed, there is another ticket for the saved searches' changes not being saved, @kocio-pl's issue #1048 :)
